lib/lte_link_control/lte_lc.c

```diff
@@ -265,7 +265,9 @@ static int parse_cereg(const char *notification,
 
 	*reg_status = status;
 
-	if (*reg_status != LTE_LC_NW_REG_UICC_FAIL) {
+	if ((*reg_status != LTE_LC_NW_REG_UICC_FAIL) &&
+	    (at_params_valid_count_get(&resp_list) >=
+	     AT_CEREG_CELL_ID_INDEX)) {
```

Comment:
This probably works fine (both TAC and cell ID are either present or absent), but to be sure that both can be parsed successfully, shouldn't the count be at least AT_CEREG_CELL_ID_INDEX + 1?

Reply:
My understanding is that the first parameter in the parameter list is `+CEREG`, so that the actual response parameters can be considered as 1-indexed

Reply:
Yes, but for example if the last parameter index is 1, the count is 2, so it's always +1.

Reply:
👍 Changed the comparison to `>`